x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/planner/QueryTranslator.java

```diff
@@ -94,17 +94,12 @@ public static void checkInsensitiveComparison(InsensitiveBinaryComparison bc) {
                 Expressions.name(bc.right()), bc.symbol());
         }
 
-        private static Query translate(InsensitiveBinaryComparison bc, TranslatorHandler handler) {
+        private static Query translate(InsensitiveBinaryComparison bc, FieldAttribute field, TranslatorHandler handler) {
```

Comment:
I tend not to agree with the additional FieldAttribute parameter. It bothers me that the methods receive a redundant parameter. In this case `bc.left()` is the same the `field` parametere.
[Here](https://github.com/elastic/elasticsearch/pull/76424/files#diff-385222df7bc48aa29c7282fd51bd3f3de071de85974214e19a44670750637182R238) `field` is the same as `isNull.field()`. This adds confusion: what's the point of the `field` if that same value can be extracted from the first parameter. Just to save on `if (X.field() instanceof FieldAttribute)`?

I'd like to know what others think about this. @bpintea @costin

Reply:
I remember I also looked longer at this. I'd say that if passing the extra parameter can be avoided everywhere, then it should be avoided.